trunk fails to build in AIX

Building the trunk fails in AIX with the following errors

builds/unix/ftsystem.c: In function 'ft_close_stream_by_munmap':
builds/unix/ftsystem.c:197:5: warning: implicit declaration of function 'munmap' [-Wimplicit-function-declaration]
     munmap( (MUNMAP_ARG_CAST)stream->descriptor.pointer, stream->size );
     ^~~~~~
builds/unix/ftsystem.c: In function 'FT_Stream_Open':
builds/unix/ftsystem.c:234:18: error: storage size of 'stat_buf' isn't known
     struct stat  stat_buf;
                  ^~~~~~~~
builds/unix/ftsystem.c:241:12: warning: implicit declaration of function 'open'; did you mean 'fopen'? [-Wimplicit-function-declaration]
     file = open( filepathname, O_RDONLY );
            ^~~~
            fopen
builds/unix/ftsystem.c:241:32: error: 'O_RDONLY' undeclared (first use in this function)
     file = open( filepathname, O_RDONLY );

This happens during the freetype build.
I am able to overcome this issue by making use of system freetype setting via setup.cfg

AIX build mechanism, unlike linux or other well tested opensource platforms, is very unique and needs special compiler flags.
So the technique of downloading freetype source code and building it just using the configure is not going to work in AIX.

The point here is once there is a new matplotlib release and when users start using "pip install matplotlib" , AIX users will face this issue. So i would probably suggest for AIX platform, matplotlib has to rely on system softwares ( in this case it's freetype)

--- a/INSTALL.rst
+++ b/INSTALL.rst
@@ -198,12 +198,13 @@ etc., you can install the following:
 FreeType and Qhull
 ------------------
 
-Matplotlib depends on `FreeType <https://www.freetype.org/>`_ (>= 2.3), a
-font rendering library, and on `Qhull <http://www.qhull.org/>`_ (>= 2015.2),
-a library for computing triangulations.  By default, Matplotlib downloads and
-builds its own copy of FreeType (this is necessary to run the test suite,
-because different versions of FreeType rasterize characters differently), and
-uses its own copy of Qhull.
+Matplotlib depends on `FreeType <https://www.freetype.org/>`_ (>=
+2.3), a font rendering library, and on `Qhull
+<http://www.qhull.org/>`_ (>= 2015.2), a library for computing
+triangulations.  By default (except on AIX) Matplotlib downloads and
+builds its own copy of FreeType (this is necessary to run the test
+suite, because different versions of FreeType rasterize characters
+differently), and uses its own copy of Qhull.
 
 To force Matplotlib to use a copy of FreeType or Qhull already installed in
 your system, create a :file:`setup.cfg` file with the following contents:

--- a/setupext.py
+++ b/setupext.py
@@ -137,8 +137,8 @@ if os.path.exists(setup_cfg):
     config.read(setup_cfg)
 options = {
     'backend': config.get('rc_options', 'backend', fallback=None),
-    'system_freetype': config.getboolean('libs', 'system_freetype',
-                                         fallback=False),
+    'system_freetype': config.getboolean(
+        'libs', 'system_freetype', fallback=sys.platform.startswith('aix')),
     'system_qhull': config.getboolean('libs', 'system_qhull',
                                       fallback=False),
 }

